Assets/Scripts/Utilities/CCLogger.cs

```diff
@@ -41,20 +48,191 @@ private static string getSceneName()
+    public static readonly string EVENT_ADD_INTERACTION = "Interaction marker Added";
+    public static readonly string APP_NAME = "CEASAR";
+
+
+    public string application;
+    public string activity;
+    public string scene;
+    public double time; // epoc millies
+    public string platformName;
+    public string username;
+    public string groupName;
+    public string message;
+
+    // event is a reserved word, but we need to send it as JSON.
+    // see: https://stackoverflow.com/questions/2787716/use-the-long-reserved-word-as-a-variable-name-in-c-sharp
+    public string @event = "nothing";
+    public string event_value;
+
+    public LatLng lastLocation;
+    public DateTime lastSimulationTime;
+    public Vector3 lastHeading;
+    public string lastLocationName;
+    public string selectedObject;
+    public string selectedStar;
+
     private static CCLogger instance;
 
     public static CCLogger GetInstance()
     {
         return instance ?? (instance = new CCLogger());
     }
 
+    private CCLogger()
+    {
+        InitializeContext();
+        UpdateContext();
+        RegisterEventListeners();
+    }
+
+    // Initialize local variables to default values.
+    // Will be updated with real data later after events start coming in.
+    private void InitializeContext()
+    {
+        application = APP_NAME;
+        activity = "activity";
+        scene = "scene";
+        platformName = "unknown";
+        username = "unknown";
+        groupName = "unknown";
+        message = "";
+        event_value = "";
+        lastLocationName = "";
+        selectedObject = "";
+        selectedStar = "";
+        lastLocation = new LatLng();
+        lastSimulationTime = new DateTime();
+        lastHeading = new Vector3();
+    }
+
+    private void UpdateContext()
+    {
+        SimulationManager manager = SimulationManager.GetInstance();
+        time = getTimeMillies();
+        scene = getSceneName();
+        platformName = Application.platform.ToString();
+        username = PlayerPrefs.GetString(SimulationConstants.USERNAME_PREF_KEY);
+        groupName = PlayerPrefs.GetString(SimulationConstants.USER_GROUP_PREF_KEY);
+        lastLocation = manager.Currentlocation;
+        lastSimulationTime = manager.CurrentSimulationTime;
+        if (manager.CurrentLocationName != null && manager.CurrentLocationName.Length > 0)
+        {
+            lastLocationName = manager.CurrentLocationName;
+        }
+        if (manager.CelestialSphereObject != null)
+        {
+            selectedObject = manager.CelestialSphereObject.name;
+        }
+        else
+        {
+            selectedObject = "";
+        }
+        if (manager.CurrentlySelectedStar != null)
+        {
+            selectedStar = manager.CurrentlySelectedStar.name;
+        }
+        else
+        {
+            selectedObject = "";
+        }
+
+    }
+
+
+    private void RegisterEventListeners()
```

Comment:
We will need to remove listeners at the end of running the application - Unity is really sticky when it comes to listeners.